addons/mail/static/src/models/message_action_list/message_action_list.js

```diff
@@ -40,6 +40,9 @@ function factory(dependencies) {
          * @param {MouseEvent} ev
          */
         onClickConfirmDelete(ev) {
+            if (this.message.isStarred) {
+                this.message.toggleStar();
+            }
```

Comment:
I wonder, isn't it a bit too much flow-specific? Let's imagine we add a new way to delete messages that doesn't go through the `DeleteMessageConfirmDialog`, the starred counter wouldn't be updated there unless we duplicate the code in the new flow.

I think it would be better to have an `OnChange` or something like that on the `body` field of Message that unstars the message when the body is empty (as this is the heuristic we currently use to determine the message has been deleted).